plugin/trino-kudu/src/main/java/org/apache/kudu/client/KuduOperationApplier.java

```diff
@@ -13,28 +13,111 @@
  */
 package org.apache.kudu.client;
 
+import io.trino.plugin.kudu.KuduClientSession;
+import io.trino.plugin.kudu.KuduClientWrapper;
 import io.trino.spi.TrinoException;
 
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkState;
 import static io.trino.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
 import static java.lang.String.format;
+import static org.apache.kudu.client.SessionConfiguration.FlushMode.MANUAL_FLUSH;
 
 /**
- * Operation.getChangeType() is package private
+ * This class is used to buffer operations and apply them in a batch and verify the batch was successfully applied.
+ * This gives:
+ * - performance of not flushing after every operation
+ * - correctness of not flushing in the background and hoping operations succeed
+ * The buffer is flushed when:
+ * - The KuduOperationApplier is closed
+ * - Or if the KuduOperationApplier reaches the bufferMaxOperations
+ * Note: Operation.getChangeType() is package private
  */
 public final class KuduOperationApplier
+        implements AutoCloseable
 {
-    private KuduOperationApplier()
+    private static final int bufferMaxOperations = 1000;
```

Comment:
Do we know how much heap usage growth we see when buferring all 1000 operations? i.e. is 1000 a very big value?

Can we add a configuration for this as a safety switch?

Or is not large enough to care about it at all?

Reply:
The default buffer size is 1000: https://github.com/apache/kudu/blob/64619147e7932518bb88b3f3f719d9c27cfdae68/java/kudu-client/src/main/java/org/apache/kudu/client/AsyncKuduSession.java#L127

So I assumed that was a reasonable default/would not consume too much memory

Reply:
Perfect.